src/main/scala/gitbucket/core/controller/RepositoryViewerController.scala

```diff
@@ -846,15 +846,19 @@ trait RepositoryViewerControllerBase extends ControllerBase {
     redirect(s"${repository.owner}/${repository.name}/releases")
   })
 
-  get("/:owner/:repository/archive/:name")(referrersOnly { repository =>
-    val name = params("name")
-    archiveRepository(name, repository, "")
+  get("/:owner/:repository/archive-whole-repo/*")(referrersOnly { repository =>
+    val name = params.get("splat")
+    if (name.isDefined) {
+      archiveRepository(name.get, repository, "")
+    }
   })
 
-  get("/:owner/:repository/archive/*/:name")(referrersOnly { repository =>
-    val name = params("name")
-    val path = multiParams("splat").head
-    archiveRepository(name, repository, path)
```

Comment:
When branch name includes `/`, this endpoint was called even if you want to download whole repo. This causes `NullPointerException` later.

Reply:
Is there any way to distinguish these routes without renaming?

Reply:
I thought we might be able to use the same logic as `tree` if URL has the following structure, for example:

- http://localhost:8080/root/repo/archive/release/v1.0.0?format=zip
- http://localhost:8080/root/repo/archive/release/v1.0.0/dir1/dir2?format=zip

However, this is just an idea. Since URL needs to be consistent and stable as much as possible, we need to design new URL very carefully if we will change it.